src/write.c

```diff
@@ -339,13 +337,12 @@ static void avifBackupSettings(avifEncoder * encoder)
     lastEncoder->tileRowsLog2 = encoder->tileRowsLog2;
     lastEncoder->tileColsLog2 = encoder->tileColsLog2;
     lastEncoder->speed = encoder->speed;
-    encoder->data->csOptionsUpdated = AVIF_FALSE;
 }
 
 // This function detect changes made on avifEncoder.
 // It reports if the change is valid, i.e. if any setting that can't change was changed.
-// It also sets needUpdate to true if valid changes are detected.
-static avifBool avifEncoderSettingsChanged(const avifEncoder * encoder, avifBool * needUpdate)
+// It also reports detected changes in updatedConfig.
```

Comment:
This function should sets `*updatedConfig` to 0 on entry. Otherwise, we need to document that it expects the caller to initialize `*updatedConfig` to 0.

Reply:
Done